private/savevar.m

```diff
@@ -15,6 +15,11 @@ function savevar(filename, varname, value, hashfile)
 
 eval(sprintf('%s = value;', varname));
 
+% Create path if it doesn't exist
+if ~exist(fileparts(filename))
+  mkdir(fileparts(filename))
```

Comment:
this will not make the directory if the parent directory does not yet exist. I suggest to implement it using a sub function with recursion, e.g. like

```
function mkdir_and_parent(d)
parent = fileparts(d)
if ~isfolder(parent)
  % this will recursively make all parent directories
  mkdir_and_parent(parent);
end
mkdir(d);
```

Reply:
I believe `mkdir` already does this.  " `mkdir parentFolder folderName` creates `folderName` in `parentFolder`. If `parentFolder` does not exist, MATLAB attempts to create it.
https://www.mathworks.com/help/matlab/ref/mkdir.html#:~:text=If%20parentFolder%20does%20not%20exist,error%20to%20the%20Command%20Window.

Reply:
you are right
```
>> mkdir('/tmp/a/b/c/d')
>> ls -al /tmp/a/b/c/d
total 0
drwxr-xr-x  2 roboos  wheel  64 May 10 16:20 .
drwxr-xr-x  3 roboos  wheel  96 May 10 16:20 ..
```